tests/test_prarallel_limiter.py

```diff
@@ -8,6 +8,6 @@ def test_parallel_limiter(flask_app):
     user = create_new_user()
     with flask_app.test_request_context():
         login_user(user)
-        pl = _InnerLock("test", max_wait_secs=1, wait_loop_secs=0.1)
+        pl = _InnerLock("test", max_wait_secs=1)
```

Comment:
can we test a case that raises the TooManyRequest exception? For example by having a slow operation in the wrapped function.

Reply:
I tried to but didn't manage to have two parallel threads running with pytest. It seems to make threads work serially. Either that or `time.sleep` doesn't release the GIL.